drivers/adc/adc_stm32.c

```diff
@@ -1212,15 +1086,18 @@ static int adc_stm32_read_async(const struct device *dev,
 }
 #endif
 
-static int adc_stm32_check_acq_time(uint16_t acq_time)
+static int adc_stm32_check_acq_time(const struct device *dev, uint16_t acq_time)
 {
+	const struct adc_stm32_cfg *config =
+		(const struct adc_stm32_cfg *)dev->config;
+
 	if (acq_time == ADC_ACQ_TIME_MAX) {
-		return ARRAY_SIZE(acq_time_tbl) - 1;
+		return STM32_NB_SAMPLING_TIME - 1;
 	}
 
-	for (int i = 0; i < 8; i++) {
+	for (int i = 0; i < STM32_NB_SAMPLING_TIME; i++) {
```

Comment:
I know the existing code works this way, so no need to be part of this PR, but it seems to me that putting this loop _after_ the simple check for `ADC_ACQ_TIME_DEFAULT` below it would probably be more efficient.  Default time being a common case and looping through the entire sampling time array to not find the time being a lot slower than a simple check against a constant.

Reply:
Done, Moved the default case at the start and the loop last